### DISCUSSION TITLE
running CUDA on mac

### DISCUSSION THREAD

**Discussion by vinayakarsh** [karpathy/llm.c#219] [Q&A]
im new to CUDA.. how to run CUDA programs on mac

**Comment by vinayakarsh** [comment]
duplicate of https://github.com/karpathy/llm.c/discussions/220#discussion-6544679

**Comment by karpathy** [comment]
You don't, you'd have to reach for Metal. This is not currently "in scope" for this repo, but maybe one of the notable ports is a place to look.

**Reply by vinayakarsh** [reply]
Thank you

**Comment by regrettable-username** [comment]
@vinayakarsh 
I'm working on a fork that's using Metal over at https://github.com/regrettable-username/llm.metal I've got the forward pass working. I'm in the process of implementing the backward pass. I've been taking a bit of a detour to set up some testing infrastructure similar to what's in dev/cuda in this repo. This will help benchmark and experiment with kernels for each layer as well as implement the layers for the backward pass.

There is also Accelerate.framework which might be used to get things running pretty quick on the CPU side. If you're on Intel based mac, that would definitely be a worthwhile pursuit!
